### PR TITLE
GA Migration Java Postcommit Jpms

### PR DESCRIPTION
**Pull request by andoni-guzman**
This PR is part of the migration from jenkins to github actions.

This PR contains this tests

Java PostCommit JPMS Dataflow - Java 11 and 17
Java PostCommit JPMS Direct - Java 11 and 17
Java PostCommit JPMS Flink - Java 11
Java PostCommit JPMS Spark - Java 11

4 Files were added to the workflows folder of github actions
6 Files were deleted, this files were the groovy files for jenkins
Jenkins readme was modified deleting migrated tests.

Waiting for this PR before merge https://github.com/apache/beam/pull/22703 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: .github/workflows/job-postcommit-java-jpms-flink.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    name: Run PostCommit Java JPMS Flink - Java 11
+    runs-on: [self-hosted, ubuntu-20.04]
+    timeout-minutes: 240
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
```
Comment (damccorm) [inline .github/workflows/job-postcommit-java-jpms-flink.yml:39]: Should this have the same todo about enabling java 17?

File: .github/workflows/job-postcommit-java-jpms-spark.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
```
Comment (damccorm) [inline .github/workflows/job-postcommit-java-jpms-spark.yml:43]: Same question

File: .github/workflows/job-postcommit-java-jpms-flink.yml (hunk 1 above)
Comment (andoni-guzman) [reply .github/workflows/job-postcommit-java-jpms-flink.yml:39]: HI Danny, No in this cases ( flink and spark) the original groovy test only contemplate Java 11

File: .github/workflows/job-postcommit-java-jpms-spark.yml (hunk 2 above)
Comment (andoni-guzman) [reply .github/workflows/job-postcommit-java-jpms-spark.yml:43]: HI Danny, No in this cases ( flink and spark) the original groovy test only contemplate Java 11

File: .github/workflows/job-postcommit-java-jpms-flink.yml (hunk 1 above)
Comment (damccorm) [reply .github/workflows/job-postcommit-java-jpms-flink.yml:39]: SGTM, thanks for clarifying

**Review by damccorm — APPROVED** [review]

**Pull request closed without merging**